api/invite_test.go

```diff
@@ -117,68 +116,64 @@ func (ts *InviteTestSuite) TestInvite_WithoutAccess() {
 }
 
 func (ts *InviteTestSuite) TestVerifyInvite() {
-	user, err := models.NewUser(ts.instanceID, "test@example.com", "", ts.Config.JWT.Aud, nil)
-	now := time.Now()
-	user.InvitedAt = &now
-	user.EncryptedPassword = ""
-	user.ConfirmationToken = "asdf"
-	require.NoError(ts.T(), err)
-	require.NoError(ts.T(), ts.API.db.Create(user))
-
-	// Find test user
-	u, err := models.FindUserByEmailAndAudience(ts.API.db, ts.instanceID, "test@example.com", ts.Config.JWT.Aud)
-	require.NoError(ts.T(), err)
-
-	// Request body
-	var buffer bytes.Buffer
-	require.NoError(ts.T(), json.NewEncoder(&buffer).Encode(map[string]interface{}{
-		"type":     "signup",
-		"token":    u.ConfirmationToken,
-		"password": "testing",
-	}))
-
-	// Setup request
-	req := httptest.NewRequest(http.MethodPost, "http://localhost/verify", &buffer)
-	req.Header.Set("Content-Type", "application/json")
-
-	// Setup response recorder
-	w := httptest.NewRecorder()
-
-	ts.API.handler.ServeHTTP(w, req)
-
-	assert.Equal(ts.T(), http.StatusOK, w.Code, w.Body.String())
-}
+	cases := []struct {
+		desc        string
+		email       string
+		requestBody map[string]interface{}
+		expected    int
+	}{
+		{
+			"Verify invite with password",
+			"test@example.com",
+			map[string]interface{}{
+				"type":     "invite",
+				"token":    "asdf",
+				"password": "testing",
+			},
+			http.StatusOK,
+		},
+		{
+			"Verify invite with no password",
+			"test1@example.com",
+			map[string]interface{}{
+				"type":  "invite",
+				"token": "asdf",
+			},
+			http.StatusOK,
```

Comment:
How come this isn't http.StatusUnprocessableEntity?

Reply:
according to the `signupVerify` method, if the invited user doesn't have a password set, it still signs the user up with a random generated password (https://github.com/supabase/gotrue/blob/ceba0d5991b5230f25187baca9592cce576c5ca3/api/verify.go#L169). This probably maps to the invite button on the dashboard where admins can just invite a user by just typing in an email.